Lib/fontTools/misc/roundTools.py

```diff
@@ -56,3 +56,50 @@ def roundFunc(tolerance, round=otRound):
         return round
 
     return functools.partial(maybeRound, tolerance=tolerance, round=round)
+
+
+def nearestMultipleShortestRepr(value: float, factor: float) -> str:
+    """Round to nearest multiple of factor and return shortest decimal representation.
+
+    This chooses the float that is closer to a multiple of the given factor while
+    having the shortest decimal representation (the least number of fractional decimal
+    digits).
+
+    For example, given the following:
+
+    >>> nearestMultipleShortestRepr(-0.61883544921875, 1.0/(1<<14))
+    '-0.61884'
+
+    Useful when you need to serialize or print a fixed-point number (or multiples
+    thereof, such as F2Dot14 fractions of 180 degrees in COLRv1 PaintRotate) in
+    a human-readable form.
+
+    Args:
+        value (value): The value to be rounded and serialized.
+        factor (float): The value which the result is a close multiple of.
+
+    Returns:
+        str: A compact string representation of the value.
+    """
+    if not value:
+        return "0.0"
+
+    value = otRound(value / factor) * factor
+    eps = .5 * factor
+    lo = value - eps
+    hi = value + eps
+    # If the range of valid choices spans an integer, return the integer.
+    if int(lo) != int(hi):
+        return str(float(round(value)))
```

Comment:
why do you need float()?

Reply:
this was intentional. I need the return value to be formatted as a float (180.0 instead of 180), this was the convention so far in ttx dumps for fields that are encoded as fixed-point values. It was already like this in the original code (from which this function is factored out) https://github.com/fonttools/fonttools/blob/12d66ded1230c46a0e62afedfe0a16bd99d78105/Lib/fontTools/misc/fixedTools.py#L145